pydantic/main.py

```diff
@@ -106,106 +101,102 @@ def __new__(
         Returns:
             type: the new class created by the metaclass
         """
-        if _base_class_defined:
-            base_field_names, class_vars, base_private_attributes = _collect_bases_data(bases)
+        # Note `ModelMetaclass` refers to `BaseModel`, but is also used to *create* `BaseModel`, so we need to keep
+        # track of whether we've created the `BaseModel` class yet, and therefore whether it's safe to refer to it. If
+        # it *hasn't* been created, we assume that the `__new__` call we're in the middle of is for the `BaseModel`
+        # class, since that's defined immediately after the metaclass.
+        if bases == (_BaseModelBase,):
```

Comment:
can we change this to `if bases != ...` and then revert most of these changes here?